### ISSUE TITLE
Error "missing :adapter configuration" on compilable project

### ISSUE THREAD

**Issue by ringods** [JakeBecker/elixir-ls#5]
I am using ElixirLS 0.0.7 on VSCode 1.13.1, running on MacOS Sierra 10.12.5.

I have a project that compiles correctly using `mix compile` (MIX_ENV=dev), but in VSCode, I get this as the first error:

![elixirlserror](https://user-images.githubusercontent.com/77923/27203417-0fabc2bc-5226-11e7-89ba-65f49fb747c3.png)

Because of this error, I suspect ElixirLS doesn't write the BEAM file for this module, resulting in all other modules using/requiring the `Repo` module to also generate errors.

`config/config.exs` has this as `Repo` config:

```
config :my_app_name, My.App.Repo,
  adapter: Ecto.Adapters.Postgres,
  database: "my_db",
  username: "postgres",
  password: "postgres",
  hostname: "localhost",
  socket_options: [recbuf: 2097152]
```

(`my_app_name` and `My.App.Repo` are redacted values)

**Comment by JakeBecker** [comment]
I've encountered this error on Phoenix projects too. Right now, unfortunately, the automatic build and error reporting is a huge hack: https://medium.com/@JakeBeckerCode/compiler-hacks-in-elixirls-6a6f04834f66

I'm working on some changes to Elixir's compiler that should make it easier to handle the automatic builds and errors (such as [this](https://github.com/elixir-lang/elixir/pull/6224)). I'm hoping that these changes will make the whole build process more robust. That said, it might take me a little while to get around to the specific build issues. You might need to hold off on using ElixirLS until 0.1.0. I'll update back here once this bug is fixed :)

**Comment by JakeBecker** [comment]
Apologies for the wait on this issue. Thanks for bearing with me.

I've rewritten the build system after contributing some changes to Elixir 1.6. Details here: https://medium.com/@JakeBeckerCode/elixirls-0-2-better-builds-code-formatter-and-incremental-dialyzer-be70999ea3e7

If you update the VS Code plugin, this issue should be fixed, though you'll need Elixir 1.6 installed to get build diagnostics in the editor. Please open a new issue if it's still broken in some way.

Thank you!

**Comment by ringods** [comment]
@JakeBecker great work. BTW, I created a #language-server channel on the Elixir slack. Not much activity there but it would be great if you could join.
